library/common/main_interface.cc

```diff
@@ -10,30 +10,39 @@
 
 // NOLINT(namespace-envoy)
 
-static std::shared_ptr<Envoy::Engine> strong_engine_;
-static std::weak_ptr<Envoy::Engine> engine_;
 static std::atomic<envoy_stream_t> current_stream_handle_{0};
 static std::atomic<envoy_network_t> preferred_network_{ENVOY_NET_GENERIC};
 
+static Envoy::Engine* engine(envoy_engine_callbacks callbacks = {0}) {
+  static Envoy::Engine* engine = new Envoy::Engine{callbacks, preferred_network_};
+  return engine; 
+}
+
 envoy_stream_t init_stream(envoy_engine_t) { return current_stream_handle_++; }
 
 envoy_status_t start_stream(envoy_stream_t stream, envoy_http_callbacks callbacks) {
-  if (auto e = engine_.lock()) {
-    return e->httpDispatcher().startStream(stream, callbacks);
+  if (auto e = engine()) {
```

Comment:
When do we expect that this could be nil - is that only during the app shutting down? Wondering because it is a static instance above

Reply:
We actually never expect that it will be nullptr with the way the interfaces are used right now by the bridge/platform layers right now. But it's a safety check that mirrors how `lock()` was used previously, and it becomes relevant if someone is calling this interface directly (e.g. with a new platform).

Reply:
got it, makes sense